modules/nf-core/dastool/dastool/tests/main.nf.test

```diff
@@ -0,0 +1,136 @@
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ],
+                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz", checkIfExists: true)
+                ])
+                input[1] = "gff"
+                """
+            }
+        }
+    }
+
+    test("dastool dastool - bacteroides fragilis - noproteins") {
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ],
+                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz", checkIfExists: true)
+                ]).join(DASTOOL_FASTATOCONTIG2BIN.out.fastatocontig2bin)
+                .map { meta, contigs, c2b -> [ meta, contigs, [], c2b ]}
+                input[1] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.fasta_proteins.get(0).get(1)).text.contains("MKIGIITICKVNNYGAELQAFATQKKLEQMGHNAEIINYLYYKDWHFKDTPLSQPFVPLD") },
+                { assert snapshot(
+                    process.out.summary,
+                    process.out.contig2bin,
+                    process.out.eval,
+                    process.out.bins,
+                    process.out.pdfs,
+                    process.out.candidates_faa,
+                    process.out.fasta_archaea_scg,
+                    process.out.fasta_bacteria_scg,
+                    process.out.b6,
+                    process.out.versions
+                ).match() }
+            )
+        }
+
+    }
+
+    test("dastool dastool - bacteroides fragilis - proteins") {
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ],
+                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz", checkIfExists: true)
+                ])
+                .join(PRODIGAL.out.amino_acid_fasta)
+                .join(DASTOOL_FASTATOCONTIG2BIN.out.fastatocontig2bin)
+                input[1] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.summary,
+                    process.out.contig2bin,
+                    process.out.eval,
+                    process.out.bins,
+                    process.out.pdfs,
+                    process.out.fasta_proteins,
+                    process.out.candidates_faa,
+                    process.out.fasta_archaea_scg,
+                    process.out.fasta_bacteria_scg,
+                    process.out.b6,
+                    process.out.versions
```

Comment:
Perhaps I'm missing something, but all of these are emitted by the process? Some of them (evals & bins) are only emitted when certain arguments are supplied, hence the lack of hashes in the snapshot.

Reply:
ok makes sense, but then if all of them are asserted we can also simply do
```
{ assert snapshot( process.out ).match() }
```

Reply:
Ah, it's because I left out `process.out.log` as it contains a timestamp, so can't match the whole process.out - have added a "contains" test now to check the contents.

Reply:
then its fine asserting the snapshots is still better then, and you can assert the log aswell with contains